frontend/src/components/NavBar.tsx

```diff
@@ -0,0 +1,120 @@
+import {
+  Box,
+  Button,
+  Flex,
+  Divider,
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverHeader,
+  PopoverBody,
+  PopoverArrow,
+  PopoverCloseButton,
+  Portal,
+  Link,
+} from "@chakra-ui/react";
+import {
+  OverViewIcon,
+  ToDoIcon,
+  PendingIcon,
+  CompletedIcon,
+  HistoryIcon,
+  SettingsIcon,
+  PlusIcon,
+} from "@/static/icons";
+import { useRouter } from "next/router";
+
+// button abstraction to include the Link and styling
+const StyleButton = ({ children, link }: { children: any; link: string }) => {
+  const router = useRouter();
+  const isActive = router.pathname === link;
+  return (
+    <>
+      <Link href={link} width="100%">
+        <Button
+          justifyContent="flex-start"
+          bg="transparent"
+          height="9"
+          width="100%"
+          borderRadius={10}
+          // temporary until I figure out custom color schemes
+          style={{
+            background: isActive ? "#FAFA78" : "white",
+          }}
+          _hover={{
+            background: "#FAFA78 !important",
```

Comment:
This might not be important, but do we want a different color for when you're simply hovering over a button versus when you've actually selected that button? Like a gray or something like that?

Reply:
This is probably a good question for design